### PR TITLE
Add QualityClient

### PR DESCRIPTION
**Pull request by AlexPerathoner**
- Implementation of the QualityClient with respective interfaces and types.
Only one function which allows to create requests for the [results/scores](https://api-docs-private.apps.k8sdev.targomo.com/#tag/Scoring-API-Private/paths/~1quality~1v1~1scores/post) of one or more locations.
- Also fixed the test:coverage command in package.json, which is now able to correctly create the html coverage report.

### CODE REVIEW

File: src/types/options/qualityRequestOptions.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+export interface QualityRequestOptions {
+  /** Criterion definitions
+   * For each criterion, a key must be set to be able to identify the different criteria in the response */
+  [criteriaId: string]: (PointOfInterestCriterion | StatisticsCriterion | GravitationalCriterion)
+}
+
+
+interface BaseCriterion {
+  /** This is the service URL to which the Targomo Reachability requests are dispatched. The endpoint has to correspond to the addresses
```
Comment (HLucasTargomo) [inline src/types/options/qualityRequestOptions.ts:11]: I would replace 'Targomo Reachability requests' by 'Targomo Routing requests' (not sure about that term though) to be more generic. poi service is using reachability requests but not statistics service (espcially when you do gravitation request).

Comment (HLucasTargomo) [reply src/types/options/qualityRequestOptions.ts:11]: but I guess this is taken from the docs.

File: src/types/types.ts [hunk 2]
```diff
@@ -885,3 +885,123 @@ export interface PoiType {
 }
 
 export type PoiHiearachy = PoiType[]
```
Comment (HLucasTargomo) [inline src/types/types.ts:887]: a small typo here: `PoiHierarchy` ;p

File: src/types/options/qualityRequestOptions.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+export interface PointOfInterestCriterion extends BaseCriterion {
+  /** must be one of these values
+  @example poiCoverageCount: the retrieved score is the number of POIs reachable from the location
+  @example poiCoverageDistance: the retrieved score is the sum of the scores weighted by the distance of each reachable POI.
+   * The higher the distance between the location and the considered POI is, the smaller is the score.
+  @example closestPoiDistance: the retrieved score is the score weighted by the distance of the closest POI reachable from the location
+    */
+  type: ('closestPoiDistance' | 'poiCoverageCount' | 'poiCoverageDistance')
+  /** list of Osm Types to consider for this criterion */
+  osmTypes: (OSMType[] | PoiGroup[])
```
Comment (HLucasTargomo) [inline src/types/options/qualityRequestOptions.ts:49]: you can actually combine OSMType and POIGroup is the same array
So I would probably put more `(OSMType | PoiGroup)[]`. I don't know if it is valid in typescript though.

File: src/types/types.ts [hunk 4]
```diff
@@ -885,3 +885,123 @@ export interface PoiType {
...
+  /** represents the id of the group */
+  value: string
+}
+
+/** See https://www.targomo.com/developers/resources/coverage/ for more infos regarding Service Coverage Areas*/
+export type CoreServiceUrl = (
+  'https://api.targomo.com/africa/' | 'https://api.targomo.com/asia/' | 'https://api.targomo.com/australia/' |
+  'https://api.targomo.com/britishisles/' | 'https://api.targomo.com/central_america/' | 'https://api.targomo.com/easterneurope/' |
+  'https://api.targomo.com/northamerica/' | 'https://api.targomo.com/south_america/' | 'https://api.targomo.com/westcentraleurope/')
+export type EndPoint = CoreServiceUrl
```
Comment (HLucasTargomo) [inline src/types/types.ts:913]: I think there are even more possible url. Some of our customers have their own core service deployment. Maybe use a simple string for this field. It doesn't have to be checked out in the client, the API(s) will take care of it.

File: src/types/types.ts [hunk 5]
```diff
@@ -885,3 +885,123 @@ export interface PoiType {
...
+
+export type Walk_Places = {
```
Comment (HLucasTargomo) [inline src/types/types.ts:915]: I think those objects exists already in this class: have a look at `LatLngIdTravelMode` `TravelSpeedValues` and `TransitTravelModeOptions`
Maybe they need to be updated and documented better though ;)

File: package-lock.json [hunk 6]
```diff
@@ -1,8 +1,10821 @@
 {
   "name": "@targomo/core",
-  "version": "0.5.0",
-  "lockfileVersion": 1,
+  "version": "0.5.1",
+  "lockfileVersion": 2,
   "requires": true,
+  "packages": {
+    "": {
```
[the remark this answers is not in the document]
Comment (AlexPerathoner) [reply package-lock.json:7]: It's not. I installed some other packages while trying to get the coverage working but forgot to remove them afterwards. All the changes to that file can be safely remove (just checked). In a moment post a commit to revert those unneeded changes.

File: src/types/options/qualityRequestOptions.ts (hunk 3 above)
Comment (AlexPerathoner) [reply src/types/options/qualityRequestOptions.ts:49]: Oh right, I misread that info. That syntax is valid.

File: src/types/types.ts [hunk 7]
```diff
@@ -884,4 +884,117 @@ export interface PoiType {
...
-export type PoiHiearachy = PoiType[]
+export type PoiHierarchy = PoiType[]
+
+
+export interface Location extends LatLngId {
+  properties?: {
+    /** Parameter used for gravitational criteria to set a custom attraction strength for the given location
+     * @default 1
+     */
+    gravitationAttractionStrength: number
```
Comment (abenrob) [inline src/types/types.ts:895]: This should be optional instead. Passing an empty object `{}` should still succeed.

**Review by HLucasTargomo — APPROVED** [review]

**Review by abenrob — APPROVED** [review]

**Pull request merged**